dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileHandler.kt

```diff
@@ -110,14 +116,33 @@ internal class BatchFileHandler(
     private fun lockFileAndWriteData(
         file: File,
         append: Boolean,
-        separator: ByteArray?,
         data: ByteArray
     ) {
         FileOutputStream(file, append).use { outputStream ->
             outputStream.channel.lock().use {
-                if (file.length() > 0 && separator != null) {
-                    outputStream.write(separator)
+                val meta = metaGenerator(data)
+
+                // in Kotlin and Java byte type is signed, meaning it goes from -127 to 128.
+                // It is completely fine to have size more than 128, it will be just stored
```

Comment:
Yes, it will be fine. It is just One's complement system https://en.wikipedia.org/wiki/Ones%27_complement. When we read byte back as `InputStream.read()` it [returns actually](https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#read()) an `int` (it applies `0xff` mask, [check here](https://github.com/openjdk/jdk/blob/82f0ac02e00f2c3ef3b10b150fcb7c7243a529bc/src/java.base/share/classes/java/io/BufferedInputStream.java#L249-L268)), so values greater than 127 which go to the negative zone in the `byte` representation will be positive if represented.

Here is an example https://ideone.com/5GJksf.

Reply:
Good to know, was not aware of that, thanks for the explanation.